src/main/java/io/nats/client/impl/SocketDataPort.java

```diff
@@ -187,4 +183,70 @@ protected static boolean isWebsocketScheme(String scheme) {
         return "ws".equalsIgnoreCase(scheme) ||
             "wss".equalsIgnoreCase(scheme);
     }
+
+    /**
+     * Implements the "Happy Eyeballs" algorithm as described in RFC 6555,
+     * which attempts to connect to multiple IP addresses in parallel to reduce
+     * connection setup delays.
+     */
+    private Socket connectToFastestIp(Options options, String hostname, int port,
+                                      int timeoutMillis) throws IOException {
+        // Get all IP addresses for the hostname
+        List<InetAddress> ips = Arrays.asList(InetAddress.getAllByName(hostname));
```

Comment:
This concerns me. I thought that ip addresses were already resolved by the time they make it here.

Reply:
There are 2 scenarios where ips are not resolved 
- when `noResolveHostnames` is enabled.
- when websocket protocol is used.
In other cases, `NatsConnection` class already does sequential tries on all ip addresses.